From 82b5eaec1176a8754997dd4f9f3c38328beeafc9 Mon Sep 17 00:00:00 2001
From: Yixun Lan <dlan@gentoo.org>
Date: Thu, 21 Aug 2025 12:15:04 +0000
Subject: [PATCH 26/33] config: spi: enable flash support

Enable SPI driver and command
Enable config for various flash devices
Enable flash test feature

Signed-off-by: Yixun Lan <dlan@gentoo.org>
---
 configs/radxa-cubie-a7a_defconfig | 13 +++++++++++++
 1 file changed, 13 insertions(+)

diff --git a/configs/radxa-cubie-a7a_defconfig b/configs/radxa-cubie-a7a_defconfig
index 76eb9e52e91..514c2c023df 100644
--- a/configs/radxa-cubie-a7a_defconfig
+++ b/configs/radxa-cubie-a7a_defconfig
@@ -12,9 +12,21 @@ CONFIG_BOOTDELAY=3
 CONFIG_SPL_RAM_DEVICE=y
 CONFIG_CMD_DFU=y
 CONFIG_CMD_I2C=y
+CONFIG_CMD_SF_TEST=y
+CONFIG_CMD_SPI=y
 CONFIG_NET_RANDOM_ETHADDR=y
 CONFIG_SYS_I2C_MVTWSI=y
 CONFIG_SUPPORT_EMMC_BOOT=y
+CONFIG_MTD=y
+CONFIG_SPI_FLASH_SOFT_RESET=y
+CONFIG_SPI_FLASH_SOFT_RESET_ON_BOOT=y
+CONFIG_SPI_FLASH_MACRONIX=y
+CONFIG_SPI_FLASH_SPANSION=y
+CONFIG_SPI_FLASH_STMICRO=y
+CONFIG_SPI_FLASH_SST=y
+CONFIG_SPI_FLASH_WINBOND=y
+CONFIG_SPI_FLASH_DATAFLASH=y
+CONFIG_SPI_FLASH_MTD=y
 CONFIG_DM_MDIO=y
 CONFIG_DM_ETH_PHY=y
 CONFIG_DWC_ETH_QOS=y
@@ -22,6 +34,7 @@ CONFIG_DWC_ETH_QOS_SUNXI=y
 CONFIG_RGMII=y
 CONFIG_RMII=y
 CONFIG_REGULATOR_AXP=y
+CONFIG_SPI=y
 # CONFIG_SYSRESET is not set
 CONFIG_USB_EHCI_HCD=y
 CONFIG_USB_OHCI_HCD=y
-- 
2.53.0

